mobile/src/components/TopNavBar.tsx

```diff
@@ -0,0 +1,46 @@
+import { Ref, useRef, useState } from "react";
+import { TextInput, View, Pressable, TextInputProps } from "react-native";
+import Icon from "react-native-vector-icons/Ionicons";
+
+interface Props extends TextInputProps {
+  value: string,
+  onChangeText: (value: string) => void;
+  focusedColor?: string;
+  unfocusedColor?: string;
+} 
+
+export default ({ value, onChangeText, focusedColor = "bg-[#C6B8F2]", unfocusedColor = "bg-[#BCC0C6]", ...props }: Props) => {
+  const inputRef: Ref<TextInput> = useRef();
+
+  const [isFocused, setFocused] = useState(false);
+
+  const bgColor = isFocused ? focusedColor : unfocusedColor;
+
```

Comment:
It might be a good idea to add some regions here

Reply:
It's only 3 lines - the regions would be more code than the actual code!

Reply:
I disagree. You should actually have boilerplate of 80 lines of all possible regions even if they are empty

Reply:
If it gets big enough, I'll add the regions.